packages/react/src/events.ts

```diff
@@ -3,21 +3,29 @@ import {
   HandlerFunction,
   isEventDataState,
   ServerEventDataState,
+  ServerHandlerDataState,
   StateDataState,
 } from './template'
 
 export function isServerEventDataState(obj: any): obj is ServerEventDataState {
   return isEventDataState(obj) && 'handler' in obj && typeof obj.handler === 'function'
 }
 
-export function event<T>(
+export function event<T extends any[]>(
```

Comment:
This makes sure that when you pass `event()` to a prop that has ServerEventDataState<[argA, argB]> signature, event(func) -> func will have correct type.

<img width="526" alt="Screenshot_2024-06-05_at_20 08 22" src="https://github.com/final-ui/final/assets/2464966/daa618f6-0ae2-4a63-a2e5-cc53a6713175">